pkg/nodes/enricher/enricher.go

```diff
@@ -46,3 +46,20 @@ func New(cves CVESuppressor, subsystem pkgMetrics.Subsystem) NodeEnricher {
 
 	return enricher
 }
+
+// NewWithCreator returns a new NodeEnricher instance for the given subsystem.
+// (The subsystem is just used for Prometheus metrics.)
+func NewWithCreator(cves CVESuppressor, subsystem pkgMetrics.Subsystem, fn func() (string, scanners.NodeScannerCreator)) NodeEnricher {
```

Comment:
Want to have `New` call this with the `clarify.NodeScannerCreator`? Also, can you update the godoc for `New` to make it clear it uses the StackRox Scanner?

Reply:
Thanks! I actually wanted to do that, but somehow forgot 😅

Reply:
Done in [957218a](https://github.com/stackrox/stackrox/pull/7592/commits/957218ac4c8f0c87f330f633b6ea57c98f762cbe)